unstructured/ingest/connector/mongodb.py

```diff
@@ -78,39 +90,209 @@ def to_dict(
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def _get_docs(self):
+        """Fetches all documents in a collection."""
+        from bson.objectid import ObjectId
+
+        # I don't like this here. Would prefer to get client
+        # from MongoDBSourceConnector or SimpleMongoDBConfig
+        client = self.connector_config.generate_client()
+        collection = self.connector_config.get_collection(client)
+        # MondoDB expects a list of ObjectIds (yet won't take a list comprehension for some reason
+        list_of_object_ids = []
+        for x in self.list_of_ids:
+            list_of_object_ids.append(ObjectId(x))
+        return list(collection.find({"_id": {"$in": list_of_object_ids}}))
+
+    def get_files(self):
+        documents = self._get_docs()
+        for doc in documents:
+            ingest_doc = MongoDBIngestDoc(
+                processor_config=self.processor_config,
+                read_config=self.read_config,
+                connector_config=self.connector_config,
+                document_meta=MongoDBDocumentMeta(
+                    collection=self.connector_config.collection,
+                    document_id=str(doc.get("_id")),
+                    date_created=doc.get("_id").generation_time.isoformat(),
+                ),
+                document=doc,
+            )
+            ingest_doc.update_source_metadata()
+            del doc["_id"]
+            filename = ingest_doc.filename
+            flattened_dict = flatten_dict(dictionary=doc)
+            str_values = [str(value) for value in flattened_dict.values()]
+            concatenated_values = "\n".join(str_values)
+
+            filename.parent.mkdir(parents=True, exist_ok=True)
+            with open(filename, "w", encoding="utf8") as f:
+                f.write(concatenated_values)
+
+            self.ingest_docs.append(ingest_doc)
+
+
+@dataclass
+class MongoDBSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    connector_config: SimpleMongoDBConfig
     _client: t.Optional["MongoClient"] = field(init=False, default=None)
 
+    @property
+    def client(self) -> "MongoClient":
+        if self._client is None:
+            self._client = self.connector_config.generate_client()
+        return self._client
+
+    def check_connection(self):
+        try:
+            self.client.admin.command("ping")
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    def initialize(self):
+        _ = self.client
+
     @requires_dependencies(["pymongo"], extras="mongodb")
-    def generate_client(self) -> "MongoClient":
-        from pymongo import MongoClient
-        from pymongo.server_api import ServerApi
+    def _get_doc_ids(self):
+        """Fetches all document ids in a collection."""
+        collection = self.connector_config.get_collection(self.client)
```

Comment:
Just making sure, this only brings in the ids and not the whole record?

Reply:
Yep. Just the ids. I'll add a return type.